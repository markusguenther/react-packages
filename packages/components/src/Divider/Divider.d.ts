/// <reference types="react" />
import { BackgroundColorProps, SizeProps } from 'styled-system';
export declare type DividerVariants = 'primary' | 'secondary' | 'inverse' | 'highlight';
export interface DividerProps extends SizeProps, BackgroundColorProps {
}
export declare const Divider: {
    (props: DividerProps): JSX.Element;
    defaultProps: {
        backgroundColor: string;
        width: number;
        height: string;
    };
};
