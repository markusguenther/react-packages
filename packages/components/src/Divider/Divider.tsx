import React from 'react';
import styled from 'styled-components';
import { BackgroundColorProps, variant } from 'styled-system';
import { theme } from '@t3n/theme';
import { Box, BoxProps } from '../Box/Box';

export type DividerVariants = 'primary' | 'secondary' | 'inverse' | 'highlight';

interface DividerProps extends Pick<BoxProps, 'width'>, BackgroundColorProps {}

const bgVariants = Object.keys(theme.colors.text).reduce(
  (colors, colorName) => ({
    ...colors,
    [colorName]: {
      bg: `text.${colorName}`
    }
  }),
  {}
);

const StyledBox = styled(Box)`
  border: none;

  ${variant({
    prop: 'bg',
    variants: bgVariants
  })}

  ${variant({
    prop: 'backgroundColor',
    variants: bgVariants
  })}
`;

export const Divider = (props: DividerProps) => {
  return <StyledBox {...props} as="hr" height="2px" />;
};

Divider.defaultProps = {
  backgroundColor: 'secondary',
  width: 1
};
