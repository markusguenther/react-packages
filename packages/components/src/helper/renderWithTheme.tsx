import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Theme, theme as originalTheme } from '@t3n/theme';
import 'jest-styled-components';

interface OptionalTheme {
  theme?: Theme;
}

export const renderWithTheme = (
  ui: React.ReactElement,
  { theme, ...options }: RenderOptions & OptionalTheme
) => {
  // merge over default theme

  const mergedTheme = { ...originalTheme, ...theme };

  const Wrapper: React.FC = ({ children }) => (
    <ThemeProvider theme={mergedTheme}>
      <>{children}</>
    </ThemeProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};
