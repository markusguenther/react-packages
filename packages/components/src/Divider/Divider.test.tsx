import React from 'react';
import { renderWithTheme } from '../helper/renderWithTheme';
import { Divider } from './Divider';

test('Divider in default/primary state matches snapshot', () => {
  const { container } = renderWithTheme(
    <>
      <Divider />
      <Divider variant="primary">test</Divider>
    </>,
    {}
  );

  expect(container).toMatchSnapshot();
});

test('Divider inverse state matches snapshot', () => {
  const { container } = renderWithTheme(
    <>
      <Divider variant="inverse" />
      <Divider variant="inverse">test</Divider>
    </>,
    {}
  );

  expect(container).toMatchSnapshot();
});

test('Divider with "width"-attribute matches snapshot', () => {
  const { container } = renderWithTheme(
    <>
      <Divider width="1" />
      <Divider width="0.5" />
      <Divider width="200px" />
      <Divider width="2rem" />
    </>,
    {}
  );

  expect(container).toMatchSnapshot();
});

test('Divider with custom margin matches snapshot', () => {
  const { container } = renderWithTheme(
    <>
      <Divider my={4} />
      <Divider mx={6} />
      <Divider mb="100px" />
      <Divider ml="3rem" />
    </>,
    {}
  );

  expect(container).toMatchSnapshot();
});
