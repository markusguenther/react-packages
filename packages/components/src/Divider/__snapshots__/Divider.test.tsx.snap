// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider in default/primary state matches snapshot 1`] = `
.c2 {
  font-family: "adelle-sans",sans-serif;
  font-size: 0.875rem;
  line-height: 1.5em;
  margin-left: 8px;
  margin-right: 8px;
  margin-top: 0;
  margin-bottom: 0;
}

.c0 {
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c1 {
  height: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  opacity: 0.6;
  background-color: #2a2a2a;
}

<div>
  <div
    class="c0"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c0"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
    <p
      class="c2"
    >
      test
    </p>
    <span
      class="c1"
    />
  </div>
</div>
`;

exports[`Divider inverse state matches snapshot 1`] = `
.c2 {
  font-family: "adelle-sans",sans-serif;
  font-size: 0.875rem;
  line-height: 1.5em;
  margin-left: 8px;
  margin-right: 8px;
  margin-top: 0;
  margin-bottom: 0;
}

.c0 {
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
}

.c1 {
  height: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  opacity: 0.6;
  background-color: #fff;
}

<div>
  <div
    class="c0"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c0"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
    <p
      class="c2"
    >
      test
    </p>
    <span
      class="c1"
    />
  </div>
</div>
`;

exports[`Divider with "width"-attribute matches snapshot 1`] = `
.c0 {
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c2 {
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 0.5;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c3 {
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 200px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c4 {
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c1 {
  height: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  opacity: 0.6;
  background-color: #2a2a2a;
}

<div>
  <div
    class="c0"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c2"
    display="flex"
    width="0.5"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c3"
    display="flex"
    width="200px"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c4"
    display="flex"
    width="2rem"
  >
    <span
      class="c1"
    />
  </div>
</div>
`;

exports[`Divider with custom margin matches snapshot 1`] = `
.c0 {
  margin-top: 24px;
  margin-bottom: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c2 {
  margin-left: 40px;
  margin-right: 40px;
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c3 {
  margin-bottom: 16px;
  margin-top: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c4 {
  margin-left: 3rem;
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2a2a2a;
}

.c1 {
  height: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  opacity: 0.6;
  background-color: #2a2a2a;
}

<div>
  <div
    class="c0"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c2"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c3"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
  </div>
  <div
    class="c4"
    display="flex"
    width="1"
  >
    <span
      class="c1"
    />
  </div>
</div>
`;
